test/unit/auth/auth.spec.ts

```diff
@@ -1005,6 +1036,101 @@ AUTH_CONFIGS.forEach((testConfig) => {
+      let clock: sinon.SinonFakeTimers;
+
+      // Stubs used to simulate underlying api calls.
+      const stubs: sinon.SinonStub[] = [];
+      beforeEach(() => {
+        stub = sinon.stub(FirebaseTokenVerifier.prototype, '_verifyAuthBlockingToken')
+          .resolves(decodedIdToken);
+        stubs.push(stub);
+        mockAuthBlockingToken = mocks.generateAuthBlockingToken();
+        clock = sinon.useFakeTimers(validSince.getTime());
+      });
+      afterEach(() => {
+        _.forEach(stubs, (s) => s.restore());
+        clock.restore();
+      });
+
+      it('should forward on the call to the token generator\'s verifyIdToken() method', () => {
+        // Stub getUser call.
+        const getUserStub = sinon.stub(testConfig.Auth.prototype, 'getUser');
+        stubs.push(getUserStub);
+        return auth._verifyAuthBlockingToken(mockAuthBlockingToken).then((result) => {
+          // Confirm getUser never called.
+          expect(getUserStub).not.to.have.been.called;
+          expect(result).to.deep.equal(decodedIdToken);
+          expect(stub).to.have.been.calledOnce.and.calledWith(mockAuthBlockingToken);
+        });
+      });
+
+      it('should reject when underlying idTokenVerifier._verifyAuthBlockingToken() rejects', () =>  {
+        const expectedError = new FirebaseAuthError(
+          AuthClientErrorCode.INVALID_ARGUMENT, 'Decoding Firebase Auth Blocking token failed');
+        // Restore verifyIdToken stub.
+        stub.restore();
+        // Simulate ID token is invalid.
+        stub = sinon.stub(FirebaseTokenVerifier.prototype, '_verifyAuthBlockingToken')
+          .rejects(expectedError);
+        stubs.push(stub);
+        return auth._verifyAuthBlockingToken(mockAuthBlockingToken)
+          .should.eventually.be.rejectedWith('Decoding Firebase Auth Blocking token failed');
+      });
+
+      it('should work with a non-cert credential when the GOOGLE_CLOUD_PROJECT environment variable is present', () => {
+        process.env.GOOGLE_CLOUD_PROJECT = mocks.projectId;
+
+        const mockCredentialAuth = testConfig.init(mocks.mockCredentialApp());
+
+        return mockCredentialAuth._verifyAuthBlockingToken(mockAuthBlockingToken).then(() => {
+          expect(stub).to.have.been.calledOnce.and.calledWith(mockAuthBlockingToken);
+        });
+      });
+
+      it('should work with a non-cert credential when the GCLOUD_PROJECT environment variable is present', () => {
+        process.env.GCLOUD_PROJECT = mocks.projectId;
+
+        const mockCredentialAuth = testConfig.init(mocks.mockCredentialApp());
+
+        return mockCredentialAuth._verifyAuthBlockingToken(mockAuthBlockingToken).then(() => {
+          expect(stub).to.have.been.calledOnce.and.calledWith(mockAuthBlockingToken);
+        });
+      });
+
+      it('should be fulfilled given an app which returns null access tokens', () => {
+        // verifyIdToken() does not rely on an access token and therefore works in this scenario.
```

Comment:
Should this be `_verifyAuthBlockingToken()`?

Reply:
yep copypaste, removed